jboss-eap-standalone-rhel7/azuredeploy.json

```diff
@@ -0,0 +1,285 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {    
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Linux VM user account name"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Administrator password for Linux VM user account"
+      }
+    },
+    "dnsLabelPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "DNS Label for the Public IP. Must be lowercase. It should match with the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$ or it will raise an error."
+      }
+    },
+    "eapUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "User name for EAP manager"
+      }
+    },
+    "eapPassword": {
+      "type": "string",
+      "metadata": {
+        "description": "Password for EAP manager"
+      }
+    },
+
+    "rhsmUserName": {
+      "type": "string",
+      "defaultValue": "mfentane@redhat.com",
+      "metadata": {
+        "description": "User name for Red Hat subscription manager"
+      }
+    },
+    "rhsmPassword": {
+      "type": "securestring",
+      "defaultValue": "Myr1am84",
+      "metadata": {
+        "description": "Password for Red Hat subscription  manager"
+      }
+    },
+    "eapProfile": {
+      "type": "string",
+      "metadata": {
+        "description": "EAP profile to use"
+      }
+    },
+    "sshPassPhrase": {
+      "type": "string",
+      "metadata": {
+        "description": "Pass phrase for SSH certificate"
+      }
+    }
+  },
+  "variables": {
+    "imagePublisher": "redhat",
+    "imageOffer": "rhel",
+    "imageSKU":"7.2",
+    "baseName": "[uniquestring(parameters('dnsLabelPrefix'),resourceGroup().id,uniquestring(deployment().name))]",
+    "nicName": "[concat('myNic8',variables('baseName'))]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnet1Name": "Subnet-1",
+    "subnet1Prefix": "10.0.0.0/24",
+    "publicIPAddressName": "[concat('myIP8',variables('baseName'))]",
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "[parameters('dnsLabelPrefix')]",
+    "vmSize": "Standard_F1",
+    "virtualNetworkName": "MyVNET8",
+    "frontEndNSGName": "[concat(parameters('dnsLabelPrefix'),'-nsg8')]",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "storageAccountType": "Standard_LRS",
+    "storageAccountName": "[concat('vsts8',variables('baseName'))]",
+    "subnet1Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]"
```

Comment:
remove concat, just use resourceId (and remove vnetID above)

Reply:
Could you elaborate more on this, pls

Reply:
[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnet1Name'))]

And then remove the vnetID variable it's not needed